packages/core/src/parsers/jsx.ts

```diff
@@ -853,6 +878,42 @@ export function parseJsx(
             const keepStatements = path.node.body.filter((statement) =>
               isImportOrDefaultExport(statement),
             );
+
+            const exportsOrLocalVariables = path.node.body.filter(
+              (statement) =>
+                !isImportOrDefaultExport(statement) &&
+                !isTypeOrInterface(statement) &&
+                !types.isExpressionStatement(statement),
+            );
+
+            context.builder.component.exports = exportsOrLocalVariables.reduce(
+              (pre, node) => {
+                let name, isFunction;
+                if (babel.types.isExportNamedDeclaration(node)) {
+                  name = (node.declaration as any).declarations[0].id.name;
+                  isFunction = babel.types.isFunction(
+                    (node.declaration as any).declarations[0].init,
+                  );
+                }
+                if (babel.types.isVariableDeclaration(node)) {
+                  name = (node.declarations as any)[0].id.name;
+                  isFunction = babel.types.isFunction(
+                    (node.declarations as any)[0].init,
+                  );
+                }
```

Comment:
This allows a bunch of nodes that will cause crashes. We should add checks for `isIdentifier` to the inner node


```suggestion
                if (
                  babel.types.isExportNamedDeclaration(node) &&
                  babel.types.isVariableDeclaration(node.declaration) &&
                  babel.types.isIdentifier(node.declaration.declarations[0].id)
                ) {
                  name = node.declaration.declarations[0].id.name;
                  isFunction = babel.types.isFunction(
                    node.declaration.declarations[0].init,
                  );
                } else if (
                  babel.types.isVariableDeclaration(node) &&
                  babel.types.isIdentifier(node.declarations[0].id)
                ) {
                  name = node.declarations[0].id.name;
                  isFunction = babel.types.isFunction(
                    node.declarations[0].init,
                  );
                }
```

Reply:
I'll fix it later